### PR TITLE
Fix beatmap combo colours not being applied

### PR DESCRIPTION
**Pull request by peppy**
BeatmapConverter wasn't transferring them due to some convoluted logic.

This also prepares to have skin.ini read for additional combo colour application in the near future.

### CODE REVIEW

File: osu.Game/Beatmaps/Formats/LegacyDecoder.cs [hunk 1]
```diff
@@ -40,7 +41,53 @@ protected override void ParseStreamInto(StreamReader stream, T beatmap)
...
+            var pair = SplitKeyVal(line, ':');
+
+            bool isCombo = pair.Key.StartsWith(@"Combo");
+
+            string[] split = pair.Value.Split(',');
+
+            if (split.Length != 3)
+                throw new InvalidOperationException($@"Color specified in incorrect format (should be R,G,B): {pair.Value}");
+
+            if (!byte.TryParse(split[0], out var r) || !byte.TryParse(split[1], out var g) || !byte.TryParse(split[2], out var b))
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/Formats/LegacyDecoder.cs]: shouldn't we be trimming each of these?

File: osu.Game/Beatmaps/Formats/LegacyDecoder.cs [hunk 2]
```diff
@@ -40,7 +41,53 @@ protected override void ParseStreamInto(StreamReader stream, T beatmap)
...
+        {
+            switch (section)
+            {
+                case Section.Colours:
+                    handleColours(output, line);
+                    return;
+            }
+        }
+
+        private bool hasCustomColours;
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/Formats/LegacyDecoder.cs]: I'd rather this was named `hasComboColours` so as to not conflict with `IHasCustomColours`.

**Review by smoogipoo — DISMISSED** [review]

File: osu.Game/Beatmaps/Formats/LegacyDecoder.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Beatmaps/Formats/LegacyDecoder.cs]: They aren't trimmed in stable, so I think it's safe to not do so

File: osu.Game/Beatmaps/Formats/LegacyDecoder.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Beatmaps/Formats/LegacyDecoder.cs]: oops

**Review by smoogipoo — APPROVED** [review]
lgtm

**Pull request merged**